framework/branch-key-store.md

```diff
@@ -578,38 +799,55 @@ The following encryption context keys are shared:
 - MUST have a `hierarchy-version`
 - MUST NOT have a `enc` attribute
 
-Any additionally attributes in the EncryptionContext
-of the [encrypted hierarchical key](./key-store/key-storage.md#encryptedhierarchicalkey)
-MUST be added to the encryption context.
-
-### ACTIVE Encryption Context
+### ACTIVE Branch Key Context
 
 The ACTIVE branch key is a copy of the DECRYPT_ONLY with the same `version`.
 It is structured slightly differently so that the active version can be accessed quickly.
 
-In addition to the [encryption context](#encryption-context):
+In addition to the [branch key context](#encryption-context):
 
-The ACTIVE encryption context value of the `type` attribute MUST equal to `"branch:ACTIVE"`.
-The ACTIVE encryption context MUST have a `version` attribute.
+The ACTIVE branch key context value of the `type` attribute MUST equal to `"branch:ACTIVE"`.
+The ACTIVE branch key context MUST have a `version` attribute.
 The `version` attribute MUST store the branch key version formatted like `"branch:version:"` + `version`.
 
-### DECRYPT_ONLY Encryption Context
+### DECRYPT_ONLY Branch Key Context
 
-In addition to the [encryption context](#encryption-context):
+In addition to the [branch key context](#branch-key-context):
 
-The DECRYPT_ONLY encryption context MUST NOT have a `version` attribute.
+The DECRYPT_ONLY branch key context MUST NOT have a `version` attribute.
 The `type` attribute MUST stores the branch key version formatted like `"branch:version:"` + `version`.
 
-### Beacon Key Encryption Context
+### Beacon Key Branch Key Context
+
+In addition to the [branch key context](#encryption-context):
+
+The Beacon key branch key context value of the `type` attribute MUST equal to `"beacon:ACTIVE"`.
+The Beacon key branch key context MUST NOT have a `version` attribute.
+
+## Encryption Context
 
-In addition to the [encryption context](#encryption-context):
+This section describes how the AWS KMS encryption context is built.
 
-The Beacon key encryption context value of the `type` attribute MUST equal to `"beacon:ACTIVE"`.
-The Beacon key encryption context MUST NOT have a `version` attribute.
+If the `hierarchy-version` is v1, AWS KMS encryption context MUST be same as [branch key context](#branch-key-context).
+If the `hierarchy-version` is v2, AWS KMS encryption context MUST be the [encryption context](../structures.md#encryption-context) send by users without any transformation.
+
+AWS KMS encryption context MUST be always the same encryption context send by user regardless of any variation of encryption context (i.e.: ACTIVE Encryption Context, DECRYPT_ONLY Encryption Context, Beacon Key Encryption Context and Custom Encryption Context).
```

Comment:
This line reads funny. I think I understand what we are trying to say.
But I think we need to clean it up.

Reply:
I rephrase it to 

```
If the `hierarchy-version` is v2, the encryption context send to AWS KMS MUST always match the encryption context originally sent by the user, without any variations. This requirement applies regardless of the type of encryption context being used, whether it is the ACTIVE encryption context, the DECRYPT_ONLY encryption context, the Beacon Key encryption context, or a custom encryption context.
```

lmk if you have any suggestion